model/stats/query.go

```diff
@@ -0,0 +1,284 @@
+	}
+	return nil
+}
+
+type Sort string
+
+func (s *Sort) validate() error {
+	switch *s {
+	case SortLatestFirst:
+	case SortEarliestFirst:
+	default:
+		return errors.Errorf("Invalid Sort value: %v", *s)
+	}
+	return nil
+}
+
+/////////////
+// Filters //
+/////////////
+
+// Represents parameters that allow a search query to resume at a specific point.
+// Used for pagination.
+type StartAt struct {
+	Date         time.Time
+	BuildVariant string
+	Task         string
+	Test         string
+	Distro       string
+}
+
+// Validates that the StartAt struct is valid for use with test stats.
+func (s *StartAt) validateCommon(groupBy GroupBy) error {
+	catcher := grip.NewBasicCatcher()
+	if !s.Date.Equal(util.GetUTCDay(s.Date)) {
+		catcher.Add(errors.New("Invalid StartAt Date value"))
+	}
+	switch groupBy {
+	case GroupByDistro:
+		if len(s.Distro) == 0 {
+			catcher.Add(errors.New("Missing StartAt Distro value"))
+		}
+		fallthrough
+	case GroupByVariant:
+		if len(s.BuildVariant) == 0 {
+			catcher.Add(errors.New("Missing StartAt BuildVariant value"))
+		}
+		fallthrough
+	case GroupByTask:
+		if len(s.Task) == 0 {
+			catcher.Add(errors.New("Missing StartAt Task value"))
+		}
+	}
+	return catcher.Resolve()
+}
+
+// Validates that the StartAt struct is valid for use with test stats.
+func (s *StartAt) validateForTests(groupBy GroupBy) error {
+	catcher := grip.NewBasicCatcher()
+	catcher.Add(s.validateCommon(groupBy))
+	if len(s.Test) == 0 {
+		catcher.Add(errors.New("Missing Start Test value"))
+	}
+	return catcher.Resolve()
+}
+
+// Validates that the StartAt struct is valid for use with task stats.
+func (s *StartAt) validateForTasks(groupBy GroupBy) error {
+	catcher := grip.NewBasicCatcher()
+	catcher.Add(s.validateCommon(groupBy))
+	if len(s.Test) != 0 {
+		catcher.Add(errors.New("StartAt for task stats should not have a Test value"))
+	}
+	return catcher.Resolve()
+}
+
+// Represents search and aggregation parameters when querying the test or task statistics.
+type StatsFilter struct {
+	Project    string
+	Requesters []string
+	AfterDate  time.Time
+	BeforeDate time.Time
+
+	Tests         []string
+	Tasks         []string
+	BuildVariants []string
+	Distros       []string
+
+	GroupNumDays int
+	GroupBy      GroupBy
+	StartAt      *StartAt
+	Limit        int
+	Sort         Sort
+}
+
+// Creates a StatsFilter with usual default values.
+// BuildVariants and Distros default to nil.
+// GroupNumDays defaults to 1.
+// GroupBy defaults to GroupByDistro.
+// Start defaults to nil.
+// Limit defaults to MaxQueryLimit
+// Sort defaults to SortLatestFirst.
+func NewDefaultStatsFilter(project string, requesters []string, after time.Time, before time.Time, tests []string, tasks []string) StatsFilter {
+	return StatsFilter{
+		Project:       project,
+		Requesters:    requesters,
+		AfterDate:     after,
+		BeforeDate:    before,
+		Tests:         tests,
+		Tasks:         tasks,
+		BuildVariants: nil,
```

Comment:
No need to explicitly set fields that are the zero value.

Reply:
This function has been removed.